ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -174,7 +183,9 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
         }
 
         final DataStreamRequestByteBuf request = requestRef.set((DataStreamRequestByteBuf)msg);
-        requests.read(request, ctx, proxies::getDataStreamOutput);
+
+        int index = (int)request.getStreamId() % clientNum;
```

Comment:
use streamId to mod maybe not the best choice, because when there are 1000 clients, each only send one stream to server, all the streamId maybe zero, then we can not balance the proxies.

Reply:
We may use clientId as well:
`
        int index = Math.toIntExact(((0xFFFFFFFFL & request.getClientId().hashCode()) + request.getStreamId()) % clientNum);
`

Reply:
Sure, good idea.